### PR TITLE
fix: should not trigger effects if computed value is not changed

### PR DESCRIPTION
**Pull request by Varixo**
Fixes: #4918

### CODE REVIEW

File: packages/qwik/src/core/signal/signal.ts [hunk 1]
```diff
@@ -345,15 +345,12 @@ export const triggerEffects = (
...
-        (effect as ComputedSignal<unknown> | WrappedSignal<unknown>).$invalid$ = true;
-        const previousSignal = signal;
-        try {
-          signal = effect;
-          effect.$effects$?.forEach(scheduleEffect);
-        } catch (e: unknown) {
-          logError(e);
-        } finally {
-          signal = previousSignal;
+        if (effect instanceof ComputedSignal || effect instanceof WrappedSignal) {
```
Comment (wmertens) [inline packages/qwik/src/core/signal/signal.ts:348]: Isn't this always true?

File: packages/qwik/src/core/signal/signal.ts [hunk 2]
```diff
@@ -345,15 +345,12 @@ export const triggerEffects = (
...
+          try {
+            retryOnPromise(() => effect.$invalidate$());
```
Comment (wmertens) [inline packages/qwik/src/core/signal/signal.ts:350]: Should force() not also use this API then?

**Review by wmertens — COMMENTED** [review]
Looks great, just two questions

File: packages/qwik/src/core/signal/signal.ts (hunk 1 above)
Comment (Varixo) [reply packages/qwik/src/core/signal/signal.ts:348]: right, we can just cast it

File: packages/qwik/src/core/signal/signal.ts (hunk 2 above)
Comment (Varixo) [reply packages/qwik/src/core/signal/signal.ts:350]: `force` don't use computeIfNeeded so probably not needed

**Review by wmertens — APPROVED** [review]
LGTM 🎉

**Pull request merged**